service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java

```diff
@@ -113,10 +113,18 @@
                                     structuralVariant.setSite2EntrezGeneId(fusion.getEntrezGeneId());
                                     structuralVariant.setSite2HugoSymbol(fusion.getGene().getHugoGeneSymbol());
                                 } else {
+                                    Gene site1Gene = getGene(site1GeneSymbol);
                                     Gene site2Gene = getGene(site2GeneSymbol);
-                                    if (site2Gene != null) {
-                                        structuralVariant.setSite2EntrezGeneId(site2Gene.getEntrezGeneId());
-                                        structuralVariant.setSite2HugoSymbol(site2Gene.getHugoGeneSymbol());
+                                    // we need to pick the right gene since the fusion gene has been assigned to structural variant site1
```

Comment:
`site2Gene` is used if the first condition fails. may be move this inside `else` block

Reply:
@kalletlak updated